### PR TITLE
add jdk11 build to CircleCI configuration

### PR DESCRIPTION
**Pull request by mattnworb**
This is intended to flush out any problems with compiling on Java 11.

### CODE REVIEW

**Comment by mattnworb** [comment]
It'd be nice to avoid the duplication between each jdk job, but I'm not familiar enough with CircleCI's new syntax to know if that is possible.

File: .circleci/config.yml [hunk 1]
```diff
@@ -48,10 +48,28 @@ jobs:
...
+          key: v1-deps-jdk11-{{ checksum "pom.xml" }}
+      - run: mvn dependency:go-offline clean compile || true
+      - save_cache:
+          key: v1-deps-jdk11-{{ checksum "pom.xml" }}
+          paths:
+            - ~/.m2
+      - run:
+          command: mvn verify
+          environment:
+            MAVEN_OPTS: -Xmx512m --add-modules java.xml.bind
```
Comment (mattnworb) [inline .circleci/config.yml]: `java.xml.bind` is removed from Java 11: https://issues.shibboleth.net/jira/browse/JPAR-110

**Comment by mattnworb** [comment]
The japicmp-maven-plugin is expecting java.xml.bind to be available, which was moved to a java module in Java 9, and then removed in Java 10.

```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 11.579 s
[INFO] Finished at: 2018-09-07T20:15:52Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.siom79.japicmp:japicmp-maven-plugin:0.6.1:cmp (default) on project apollo-core: Execution default of goal com.github.siom79.japicmp:japicmp-maven-plugin:0.6.1:cmp failed: A required class was missing while executing com.github.siom79.japicmp:japicmp-maven-plugin:0.6.1:cmp: javax/xml/bind/JAXBException
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>com.github.siom79.japicmp:japicmp-maven-plugin:0.6.1
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/root/.m2/repository/com/github/siom79/japicmp/japicmp-maven-plugin/0.6.1/japicmp-maven-plugin-0.6.1.jar
[ERROR] urls[1] = file:/root/.m2/repository/org/sonatype/aether/aether-util/1.11/aether-util-1.11.jar
[ERROR] urls[2] = file:/root/.m2/repository/org/sonatype/sisu/sisu-inject-bean/2.1.1/sisu-inject-bean-2.1.1.jar
[ERROR] urls[3] = file:/root/.m2/repository/org/sonatype/sisu/sisu-guice/2.9.4/sisu-guice-2.9.4-no_aop.jar
[ERROR] urls[4] = file:/root/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.14/plexus-interpolation-1.14.jar
[ERROR] urls[5] = file:/root/.m2/repository/org/codehaus/plexus/plexus-utils/2.0.6/plexus-utils-2.0.6.jar
[ERROR] urls[6] = file:/root/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.5.5/plexus-component-annotations-1.5.5.jar
[ERROR] urls[7] = file:/root/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.3/plexus-sec-dispatcher-1.3.jar
[ERROR] urls[8] = file:/root/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
[ERROR] urls[9] = file:/root/.m2/repository/com/github/siom79/japicmp/japicmp/0.6.1/japicmp-0.6.1.jar
[ERROR] urls[10] = file:/root/.m2/repository/org/javassist/javassist/3.20.0-GA/javassist-3.20.0-GA.jar
[ERROR] urls[11] = file:/root/.m2/repository/com/google/guava/guava/18.0/guava-18.0.jar
[ERROR] urls[12] = file:/root/.m2/repository/io/airlift/airline/0.7/airline-0.7.jar
[ERROR] urls[13] = file:/root/.m2/repository/com/google/code/findbugs/annotations/2.0.3/annotations-2.0.3.jar
[ERROR] urls[14] = file:/root/.m2/repository/org/apache/maven/reporting/maven-reporting-impl/2.3/maven-reporting-impl-2.3.jar
[ERROR] urls[15] = file:/root/.m2/repository/backport-util-concurrent/backport-util-concurrent/3.1/backport-util-concurrent-3.1.jar
[ERROR] urls[16] = file:/root/.m2/repository/junit/junit/3.8.1/junit-3.8.1.jar
[ERROR] urls[17] = file:/root/.m2/repository/org/apache/maven/shared/maven-shared-utils/0.6/maven-shared-utils-0.6.jar
[ERROR] urls[18] = file:/root/.m2/repository/com/google/code/findbugs/jsr305/2.0.1/jsr305-2.0.1.jar
[ERROR] urls[19] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-sink-api/1.2/doxia-sink-api-1.2.jar
[ERROR] urls[20] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-core/1.2/doxia-core-1.2.jar
[ERROR] urls[21] = file:/root/.m2/repository/xerces/xercesImpl/2.9.1/xercesImpl-2.9.1.jar
[ERROR] urls[22] = file:/root/.m2/repository/xml-apis/xml-apis/1.3.04/xml-apis-1.3.04.jar
[ERROR] urls[23] = file:/root/.m2/repository/commons-lang/commons-lang/2.4/commons-lang-2.4.jar
[ERROR] urls[24] = file:/root/.m2/repository/org/apache/httpcomponents/httpclient/4.0.2/httpclient-4.0.2.jar
[ERROR] urls[25] = file:/root/.m2/repository/org/apache/httpcomponents/httpcore/4.0.1/httpcore-4.0.1.jar
[ERROR] urls[26] = file:/root/.m2/repository/commons-codec/commons-codec/1.3/commons-codec-1.3.jar
[ERROR] urls[27] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-site-renderer/1.2/doxia-site-renderer-1.2.jar
[ERROR] urls[28] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-decoration-model/1.2/doxia-decoration-model-1.2.jar
[ERROR] urls[29] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-module-xhtml/1.2/doxia-module-xhtml-1.2.jar
[ERROR] urls[30] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-module-fml/1.2/doxia-module-fml-1.2.jar
[ERROR] urls[31] = file:/root/.m2/repository/org/codehaus/plexus/plexus-i18n/1.0-beta-7/plexus-i18n-1.0-beta-7.jar
[ERROR] urls[32] = file:/root/.m2/repository/org/codehaus/plexus/plexus-velocity/1.1.7/plexus-velocity-1.1.7.jar
[ERROR] urls[33] = file:/root/.m2/repository/org/apache/velocity/velocity/1.5/velocity-1.5.jar
[ERROR] urls[34] = file:/root/.m2/repository/oro/oro/2.0.8/oro-2.0.8.jar
[ERROR] urls[35] = file:/root/.m2/repository/commons-collections/commons-collections/3.2.1/commons-collections-3.2.1.jar
[ERROR] urls[36] = file:/root/.m2/repository/commons-validator/commons-validator/1.3.1/commons-validator-1.3.1.jar
[ERROR] urls[37] = file:/root/.m2/repository/commons-beanutils/commons-beanutils/1.7.0/commons-beanutils-1.7.0.jar
[ERROR] urls[38] = file:/root/.m2/repository/commons-digester/commons-digester/1.6/commons-digester-1.6.jar
[ERROR] urls[39] = file:/root/.m2/repository/commons-logging/commons-logging/1.0.4/commons-logging-1.0.4.jar
[ERROR] urls[40] = file:/root/.m2/repository/org/apache/maven/reporting/maven-reporting-api/2.2.1/maven-reporting-api-2.2.1.jar
[ERROR] urls[41] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-logging-api/1.1/doxia-logging-api-1.1.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------
[ERROR] : javax.xml.bind.JAXBException
[ERROR] -> [Help 1]
```

I found https://github.com/siom79/japicmp/issues/183 but sounds like its dependency has not been addressed in the plugin (last update 2017). Not sure the right combination of dependencies to add of `jaxb-api`, `jaxb-core` etc to address that, will need to research more.

**Comment by tommyulfsparre** [comment]
@mattnworb thanks for doing this!

**Pull request merged**